### DISCUSSION TITLE
Register blade - theme-coreuiv2

### DISCUSSION THREAD

**Discussion by blondie63** [Laravel-Backpack/community-forum#651] [Q&A (Help)] [answered]
I've seen that theme-coreuiv2 register.blade.php extend plain.blade.php and the first his line is:
```
<html lang="{{ app()->getLocale() }}" dir="{{ config('backpack.base.html_direction') }}">
```

I'd like to add a language selector in this register and also login.blade.php

Some help ? :)

Maybe it's also a good idea to have it default in backpack..

**Comment by tabacitu** [comment]
Hey @blondie63 ,

To be 100% certain of the scope here... the purpose of a language selector in the Register screen would be to... store in the database that a certain user prefers a certain language, correct? Then when that user is logged in, to switch the language of the interface from EN to FR for example, correct?

In that case, the steps would be to:
- create a migration to add a `language` column to the `users` table
- override the register blade file and controller to also include a language picker - you have [steps here](https://backpackforlaravel.com/docs/6.x/base-how-to#add-one-or-more-fields-to-the-register-form-1)
- create a middleware that changes the Laravel app language config, for that user preference

Hope it helps.
Cheers!

PS. Seems to me like you're using Backpack outside its primary scope, as an admin panel. You use it as a user interface. While that is perfectly fine, it's not how 90% of all devs use Backpack. Since most admin panels have one/two admins and all of them use one language, I don't thinks this makes the cut as an official feature. I would agree to having an external component... a language picker... that you could include as an add-on. I'll talk to the team to maybe do that, but that might take us some time, so please roll your own solution.

**Reply by blondie63** [reply]
Hi @tabacitu you are right, i'm using Backpack to create webapp as user interface because is easy to to do and works fine
For that reason i'd like to add language selector for Register and Login blade, i can have international users.. :)

**Comment by o15a3d4l11s2** [comment, marked as answer]
In the past (Laravel 8) I have added a Localization middleware in the `Kernel.php` - in the web `$middlewareGroups`.

The middleware itself is pretty simple (I am writing just a simple piece of it):
```
class Localization
{
    public function handle(Request $request, Closure $next)
    {
        $locale = Auth::user()?->locale ?? 'en';
        App::setLocale($locale);

        return $next($request);
    }
}
```

Then you can decide how/when the user can change the locale.

Of course, if you prefer, you could also store the locale in the current session and preset it from there.
